CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+2. **Clone your fork** locally:
+   ```bash
+   git clone https://github.com/YOUR-USERNAME/spyder-website.git
+   cd spyder-website
+   ```
+3. **Create a new branch** for your changes:
+   ```bash
+   git checkout -b your-feature-branch
+   ```
+
+## Development Setup
+
+1. **Install dependencies**:
+   ```bash
+   npm install
+   ```
+
+2. **Start the development server**:
+   ```bash
+   npm run dev
+   ```
+   The site will be available at `http://localhost:5173`
+
+3. **Available scripts**:
+   - `npm run dev` - Start development server
+   - `npm run build` - Build for production
+   - `npm run preview` - Preview production build
+
+## Types of Contributions
+
+We welcome various types of contributions:
+
+- **Blog posts** - Share news, tutorials, or insights about Spyder
+- **Bug fixes** - Fix issues in the website functionality
+- **Feature enhancements** - Improve existing features or add new ones
+- **Documentation** - Improve clarity and completeness of documentation
+
+## Creating Blog Posts
+
+### Adding New Authors
+
+If the blog post author(s) haven't been added yet:
+
+1. **Create author folder** in `static/assets/authors/` using a simple identifier (no spaces/special characters)
+2. **Add author files**:
+   - Square avatar image (recommended: 512×512px WEBP format)
+   - `metadata.json` file:
+   ```json
+   {
+       "name": "Author's Display Name",
+       "image": "filename.webp"
+   }
+   ```
+
+### Creating the Blog Post
+
+1. **Create post folder** in `src/routes/blog/` (folder name becomes URL slug - use lowercase, no spaces)
+
+2. **Add main content file** `+page.md` with proper frontmatter:
+   ```yaml
+   ---
+   title: Blog Post Title # Required
+   author: author-id  # Required. For multiple authors, use array: [author1, author2]
+   pub_date: YYYY-MM-DD # Required (ISO format)
+   category: Category Name  # Optional
+   tags: Tag1, Tag2  # Optional (comma-separated)
+   summary: Concise overview of content # Required
+   ---
+   ```
+
+3. **Write content** using mdsvex Markdown:
+   - [Never indent code blocks](https://mdsvex.pngwn.io/docs#limitations)
+   - Use standard Markdown syntax for formatting
+   - Follow our [style guide](#style-guidelines) for consistency
+
+4. **Add images** (if needed):
+   ```markdown
+   ![Meaningful description](image.webp "Optional caption")
+   ```
+   - Store images in the post's folder
+   - Use descriptive alt text (avoid "image of...", "picture of...", etc.)
+   - Prefer WEBP format for better performance
+
+5. **Preview your post**:
+   ```bash
+   npm run dev
+   ```
+   - Verify post appears at `/blog/your-post-name`
+   - Check author metadata and image display
+   - Test on different screen sizes
+
+## Code Contributions
+
+### Guidelines
+
+- Follow the existing code style and conventions
+- Write clear, descriptive commit messages
+- Test your changes thoroughly
+- Update documentation if needed
```

Comment:
This rather generic advice is or will be be made redundant by the specific reminders in the relevant sections, so no need to include it in the new integrated version